### PR TITLE
Remove excess initialization from BGC code

### PR DESCRIPTION
**Pull request by peterdschwartz**
Every model timestep, the vegetation and column fluxes are set to zero for the CNP variables.
This took up roughly 1/3rd of the time for the Ecosystem model time, and this PR reduces that by over half. 

[BFB]

### CODE REVIEW

File: components/elm/src/biogeochem/SoilLittDecompMod.F90 [hunk 1]
```diff
@@ -308,7 +308,9 @@ subroutine SoilLittDecompAlloc (bounds, num_soilc, filter_soilc,    &
                              / cp_decomp_pools_new(c,j,cascade_receiver_pool(k)) )
 
                      else   ! 100% respiration
-                        pmpf_decomp_cascade(c,j,k) = - p_decomp_cpool_loss(c,j,k) / cp_decomp_pools(c,j,cascade_donor_pool(k))
+                        if(decomp_ppools_vr(c,j,cascade_donor_pool(k)) > 0._r8) then
```
[the remark this answers is not in the document]
Comment (rgknox) [reply components/elm/src/biogeochem/SoilLittDecompMod.F90:311]: yes, sorry for the late reply!

**Review by rgknox — APPROVED** [review]
These look good to me @peterdschwartz

File: components/elm/src/data_types/ColumnDataType.F90 [hunk 2]
```diff
@@ -9063,7 +9002,7 @@ subroutine col_nf_zero_forfates_veg(this, bounds, num_soilc, filter_soilc)
     integer                , intent(in)    :: filter_soilc(:) ! filter for soil columns
     ! locals
     integer :: fc
-    integer :: c
+    integer :: c,k,j
```
Comment (bishtgautam) [inline components/elm/src/data_types/ColumnDataType.F90:9005]: These changes are also not needed.

**Comment by peterdschwartz** [comment]
addressed comments

**Review by bishtgautam — APPROVED** [review]

**Comment by peterdschwartz** [comment]
I see some of the e3sm_integration BGC tests have diffs with intel compiler only.  I'll try to have it sorted today

**Comment by rljacob** [comment]
Looks like it was a small number of fields.

**Comment by peterdschwartz** [comment]
@bishtgautam Queue times have been slow, but I've isolated the issue to a specific subroutine.  I'll be able to push a fix sometime tonight (depending on job turnover), but I wouldn't worry about being able to re-merge it tonight.

**Comment by rljacob** [comment]
Testing found the same 5 variables diffing in 2 cases.  Seen on both chrysalis and pm-cpu (The only ones running integration with baselines)

The cases:
```
SMS_Ld2.ne30pg2_r05_EC30to60E2r2.BGCEXP_CNTL_CNPECACNT_1850.*_*.elm-bgcexp
SMS_Ld2.ne30pg2_r05_EC30to60E2r2.BGCEXP_CNTL_CNPRDCTC_1850.*_*.elm-bgcexp
```

The variables (from one of the cases, grepping for "RMS" in cprnc.out)
```
 RMS x2a_Fall_fco2_lnd                5.6027E+32            NORMALIZED  1.1852E+01
 RMS a2x_Sa_co2prog                   4.2255E+41            NORMALIZED  1.3157E+01
 RMS l2x_Fall_fco2_lnd                1.9829E+33            NORMALIZED  3.6984E+00
 RMS x2l_Sa_co2prog                   1.5536E+42            NORMALIZED  4.0495E+00
 RMS x2oacc_Sa_co2prog                6.1847E+40            NORMALIZED  1.9795E+01
```
This seems like its a simple fix.

But 2 other cases had 255 variables diffing (which implies a real diff introduced in to the state early since its only 5 steps).
The cases:
```
SMS_Ln5.ne30pg2_r05_EC30to60E2r2.BGCEXP_LNDATM_CNPRDCTC_1850.*_*
SMS_Ln5.ne30pg2_r05_EC30to60E2r2.BGCEXP_LNDATM_CNPRDCTC_20TR.*_*
```

**Comment by peterdschwartz** [comment]
@rljacob Yes i am well aware of the differences. I tried to focus on 
```
SMS_Ln5.ne30pg2_r05_EC30to60E2r2.BGCEXP_LNDATM_CNPRDCTC_1850.*_*
```
as it is shorter and requires less resources, but I ran into the issue on chrysalis where this test DIFFs (in 255 variables) with master.  I asked Gautam to do the same, and his case also gave DIFFs (same 255 variables) with master, so I think something is wrong with the default chrysalis environment 

I don't have this issue with pm-cpu but the queue times are pretty long and that's the only thing holding this debugging effort back.

**Comment by rljacob** [comment]
I was just noting for the record.

The jenkins testing on Chrysalis uses different node counts then regular runs for a given resolution.  If you are comparing a run you do with the Jenkins-generated baseline then you might be picking up a node-count diff that has crept in.   Try making your own baseline with the same node count as your test case.

**Comment by peterdschwartz** [comment]
@rljacob  Right! the cases do also report NLDIFF for the pelayout so that could be it, thanks.

**Comment by rljacob** [comment]
Good that should help find the issue with this PR.  Unfortunately, there's now another issue because that case can't pass a PEM test.  That may have existed for a while and can be looked at later.

**Comment by peterdschwartz** [comment]
@bishtgautam Newest commit has fixed the 4 test diffs

**Comment by rljacob** [comment]
@bishtgautam you can merge this to master.

**Pull request merged**